### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by dori-master-game-dev**
This pull request introduces three new features to the project. First, a new file has been created that uses the observer pattern to send notifications to various parts of the application. This allows for a more modular and flexible approach to sending notifications.

Second, a new Django app called "notifications" has been created to handle the logic and storage of notifications. This app provides an easy-to-use interface for creating and managing notifications.

Finally, a new template has been created with AJAX functionality to update notifications without requiring a full page reload. This enhances the user experience by allowing for real-time updates to notifications.

Together, these features provide a comprehensive and efficient solution for handling notifications in the project. They have been thoroughly tested and are ready for implementation.

### CODE REVIEW

**Review by OsamaAbuOshaibah — APPROVED** [review]

**Review by OsamaMomani — CHANGES_REQUESTED** [review]
@dori-master-game-dev  Very nice code, I really enjoyed reviewing it.
Good job! 🚀

PS. I have added some nitpicking comments because I have to ;)

File: src/notification/notification.py [hunk 1]
```diff
@@ -1,43 +1,98 @@
-from abc import ABC, abstractmethod
 import inspect
 
 
-class Notification(ABC):
-    notification_args = {}
+notifications = {}
```
[the remark this answers is not in the document]
Comment (OsamaMomani) [reply src/notification/notification.py:4]: > You are right about it being confusing, they technically are notifications that can be called, but are also different from the one present in the database which contain data, I will try to find a more suitable name for them. Any suggestions would be appreciated.

Yea, so I worked on a similar system, specifically AWS SNS. The notification items being stored are called "messages", while the category of those messages is known as a "topic.", then a method -or a consumer- can subscribe to a topic -or list of topics-. In my opinion, this naming convention would be fitting for our particular situation.

File: src/notification/notification.py [hunk 2]
```diff
@@ -1,43 +1,98 @@
...
+    # Check if the notification exists
+    if notification not in notifications:
+        raise Exception('Notification does not exist')
+
+    # Check if the argument is callable
+    if not callable(subscriber_method):
+        raise Exception('Argument provided is not a function')
+
+    # Get the method arguments
+    args = inspect.getfullargspec(subscriber_method).args
```
[the remark this answers is not in the document]
Comment (OsamaMomani) [reply src/notification/notification.py:46]: I prefer meaningful and descriptive variable names, and I am a dictator when it comes to naming. So you better change it 😊

**Review by o54ma-4l5h4r1f — APPROVED** [review]

**Review by OsamaMomani — APPROVED** [review]
Noice work buddy

**Pull request merged**